File: KIWI/Factory/DataCollectorFactory.cs
using IDAUtil;
using IDAUtil.Service;
using lib;

namespace KIWI {
    static class DataCollectorFactory {
        public static DataCollectorSap createDataCollectorSAP(string salesOrg) {
            return new DataCollectorSap((ISAPLib)new ZV04I(Create.sapLib(), salesOrg, IDAEnum.Task.quantityConversion), Create.exportParses());
        }
    }
}

File: KIWI/Factory/DataCollectorFactory.vb
Imports IDAUtil
Imports [lib]

Module DataCollectorFactory

    Public Function createDataCollectorSAP(salesOrg As String) As DataCollectorSAP
        Return New DataCollectorSAP(
            New ZV04I(Create.sapLib,
                      salesOrg,
                      IDAEnum.Task.quantityConversion),
            Create.exportParses)
    End Function

End Module
